homeassistant/components/prometheus/__init__.py

```diff
@@ -151,10 +152,23 @@ def handle_event(self, event):
         if hasattr(self, handler):
```

Comment:
Perhaps we should add a check here to avoid calling `_handle_{domain}` functions for unavailable entities? This will resolve #35978 and #30186.